proto/noble/dollar/vaults/v2/vaults.proto

```diff
@@ -66,6 +66,51 @@ message UserPosition {
 
   // Number of active withdrawal requests
   int32 active_withdrawal_requests = 7;
+
+  // Position ID (unique per user)
+  uint64 position_id = 8;
+}
+
+// AccountingSnapshot represents a pending update to a user position during accounting.
+// These snapshots are used to stage all accounting changes before atomically committing
+// them when the accounting session completes. This ensures consistency even if accounting
+// is performed across multiple message invocations.
+message AccountingSnapshot {
+  // User address
+  string user = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+
+  // Position ID
+  uint64 position_id = 2;
```

Comment:
I think we may want a wrapping message `AccountingSnapshots` (plural) so a user can be mapped to more than one position? As it stands we can only save one position snapshot per user.

Reply:
Working on this